controllers/tc000201_auto_approve_with_disabled_drift_detection_test.go

```diff
@@ -0,0 +1,185 @@
+			},
+		},
+		URL: server.URL() + "/file.tar.gz",
+		Artifact: &sourcev1.Artifact{
+			Path:           "gitrepository/flux-system/test-tf-controller/b8e362c206e3d0cbb7ed22ced771a0056455a2fb.tar.gz",
+			URL:            server.URL() + "/file.tar.gz",
+			Revision:       "master/b8e362c206e3d0cbb7ed22ced771a0056455a2fb",
+			Checksum:       "80ddfd18eb96f7d31cadc1a8a5171c6e2d95df3f6c23b0ed9cd8dddf6dba1406", // must be the real checksum value
+			LastUpdateTime: metav1.Time{Time: updatedTime},
+		},
+	}
+	g.Expect(k8sClient.Status().Update(ctx, &testRepo)).Should(Succeed())
+
+	By("checking that the status and its URL gets reconciled")
+	gitRepoKey := types.NamespacedName{Namespace: "flux-system", Name: sourceName}
+	createdRepo := sourcev1.GitRepository{}
+	g.Expect(k8sClient.Get(ctx, gitRepoKey, &createdRepo)).Should(Succeed())
+
+	By("creating a new TF and attaching to the repo, with approve plan set to auto and drift detection disabled")
+	testEnvKubeConfigPath, err := findKubeConfig(testEnv)
+	g.Expect(err).Should(BeNil())
+
+	helloWorldTF := infrav1.Terraform{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      terraformName,
+			Namespace: "flux-system",
+		},
+		Spec: infrav1.TerraformSpec{
+			Interval:              metav1.Duration{Duration: tfSyncInterval},
+			ApprovePlan:           "auto",
+			DisableDriftDetection: true,
+			Path:                  "./terraform-hello-world-example",
+			BackendConfig: &infrav1.BackendConfigSpec{
+				SecretSuffix:    terraformName,
+				InClusterConfig: false,
+				ConfigPath:      testEnvKubeConfigPath,
+			},
+			SourceRef: infrav1.CrossNamespaceSourceReference{
+				Kind:      "GitRepository",
+				Name:      sourceName,
+				Namespace: "flux-system",
+			},
+		},
+	}
+	g.Expect(k8sClient.Create(ctx, &helloWorldTF)).Should(Succeed())
+
+	By("checking that the hello world TF gets created")
+	helloWorldTFKey := types.NamespacedName{Namespace: "flux-system", Name: terraformName}
+	createdHelloWorldTF := infrav1.Terraform{}
+	g.Eventually(func() bool {
+		err := k8sClient.Get(ctx, helloWorldTFKey, &createdHelloWorldTF)
+		if err != nil {
+			return false
+		}
+		return true
+	}, timeout, interval).Should(BeTrue())
+
+	By("checking that the planned status of the TF program is created successfully")
+	g.Eventually(func() map[string]interface{} {
+		err := k8sClient.Get(ctx, helloWorldTFKey, &createdHelloWorldTF)
+		if err != nil {
+			return nil
+		}
+		for _, c := range createdHelloWorldTF.Status.Conditions {
+			if c.Type == "Plan" {
+				return map[string]interface{}{
+					"Type":    c.Type,
+					"Reason":  c.Reason,
+					"Pending": createdHelloWorldTF.Status.Plan.Pending,
+					"Message": c.Message,
+				}
+			}
+		}
+		return nil
+	}, timeout, interval).Should(Equal(map[string]interface{}{
+		"Type":    "Plan",
+		"Reason":  "TerraformPlannedWithChanges",
+		"Pending": "plan-master-b8e362c206e3d0cbb7ed22ced771a0056455a2fb",
```

Comment:
Please remove this check, because sometime the state is already moved to apply, and the pending plan is already clear.